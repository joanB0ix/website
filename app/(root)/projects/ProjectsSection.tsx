import { Section } from "@/components/Section";
import { Project } from "@/config/projectList";

export interface ProjectSectionProps {
  projectList: Project[];
}

export default function ProjectsSection({ projectList }: ProjectSectionProps) {
  return (
    <Section title="My Projects">
      <ul className="list-disc pl-6">
        {projectList.map((project) => {
          return (
            <li key={project.title}>
              <span className="underline">{project.title}</span>{" "}
              {project.description}
            </li>
          );
        })}
      </ul>
    </Section>
  );
}
