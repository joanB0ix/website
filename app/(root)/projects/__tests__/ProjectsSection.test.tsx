import { render, screen } from "@testing-library/react";
import ProjectsSection, { ProjectSectionProps } from "../ProjectsSection";

const PROJECT_LIST: ProjectSectionProps["projectList"] = [
  {
    title: "Project One",
    description: "Description for Project One.",
  },
  {
    title: "Project Two",
    description: "Description for Project Two.",
  },
];

describe("ProjectsSection", () => {
  it("Should render correctly", () => {
    render(<ProjectsSection projectList={PROJECT_LIST} />);

    const sectionTitle = screen.getByText("My Projects");
    expect(sectionTitle).toBeInTheDocument();

    PROJECT_LIST.forEach((project) => {
      const projectTitle = screen.getByText(project.title);
      expect(projectTitle).toBeInTheDocument();

      const projectDescription = screen.getByText(project.description);
      expect(projectDescription).toBeInTheDocument();
    });
  });

  it("Should not render list when there are no projects", () => {
    render(<ProjectsSection projectList={[]} />);
    const listElement = screen.queryByRole("list");
    expect(listElement).toBeEmptyDOMElement();
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(
      <ProjectsSection projectList={PROJECT_LIST} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
