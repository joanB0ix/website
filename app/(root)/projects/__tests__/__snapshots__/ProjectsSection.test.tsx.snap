// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsSection Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-6"
  >
    <h2
      class="font-bold pb-2"
    >
      My Projects
    </h2>
    <ul
      class="list-disc pl-6"
    >
      <li>
        <a
          class="underline"
          href="url1"
        >
          Project One
        </a>
         Description for Project One.
      </li>
      <li>
        <a
          class="underline"
          href="url2"
        >
          Project Two
        </a>
         Description for Project Two.
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
